### PR TITLE
Don't assume plugin-whitelisted attributes are proc macro attributes

### LINKED ISSUE #40001: Custom attributes from unstable plugins and procedural macro attributes are not compatible
In https://github.com/servo/servo/compare/master...nox:custom-derive I changed the attribute `#[dom_struct]` from our `plugins` crate (a `plugin = true` crate) into a `proc_macro_attribute` function. Unfortunately, the `plugins` crate also defines a whitelisted attribute `#[must_root]` and using both at the same time fails with the following compile error:

```
error: macro undefined: `must_root`
   --> /Users/nox/src/servo/components/script/serviceworkerjob.rs:115:1
    |
115 | #[must_root]
    | ^^^^^^^^^^^^
```

### PR DESCRIPTION
**Pull request by abonander**
closes #40001

### CODE REVIEW

**Comment by abonander** [comment]
r? @jseyfried 
cc @keeperofdakeys

**Comment by jseyfried** [comment]
Excellent, thanks!
@bors r+

File: src/librustc_resolve/macros.rs [hunk 1]
```diff
@@ -181,6 +181,11 @@ impl<'a> base::Resolver for Resolver<'a> {
     fn find_legacy_attr_invoc(&mut self, attrs: &mut Vec<ast::Attribute>)
                               -> Option<ast::Attribute> {
         for i in 0..attrs.len() {
+            if self.session.plugin_attributes.borrow().iter()
+                    .any(|&(ref attr_nm, _)| attrs[i].check_name(attr_nm)) {
```
Comment (jseyfried) [inline src/librustc_resolve/macros.rs]: `check_name` marks the attribute as used when it returns `true`, which is breaking `plugin-attr-register-deny.rs` on Travis.

I would use `attrs[i].name() == attr_nm` instead.

**Comment by nox** [comment]
@Manishearth Could this get a p=1?

**Pull request merged**